stabilizer/src/main/java/com/hazelcast/stabilizer/provisioner/AwsProvisioner.java

```diff
@@ -0,0 +1,333 @@
+import com.amazonaws.services.ec2.model.RunInstancesResult;
+import com.amazonaws.services.ec2.model.TerminateInstancesRequest;
+import com.amazonaws.services.ec2.model.TerminateInstancesResult;
+import com.amazonaws.services.elasticloadbalancing.AmazonElasticLoadBalancingClient;
+import com.amazonaws.services.elasticloadbalancing.model.CreateLoadBalancerRequest;
+import com.amazonaws.services.elasticloadbalancing.model.CreateLoadBalancerResult;
+import com.amazonaws.services.elasticloadbalancing.model.DescribeLoadBalancersRequest;
+import com.amazonaws.services.elasticloadbalancing.model.DescribeLoadBalancersResult;
+import com.amazonaws.services.elasticloadbalancing.model.Listener;
+import com.amazonaws.services.elasticloadbalancing.model.LoadBalancerDescription;
+import com.amazonaws.services.elasticloadbalancing.model.RegisterInstancesWithLoadBalancerRequest;
+import com.amazonaws.services.elasticloadbalancing.model.RegisterInstancesWithLoadBalancerResult;
+import com.hazelcast.stabilizer.common.AgentAddress;
+import com.hazelcast.stabilizer.common.AgentsFile;
+import com.hazelcast.stabilizer.common.StabilizerProperties;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+
+import static com.hazelcast.stabilizer.Utils.appendText;
+import static com.hazelcast.stabilizer.Utils.sleepMillis;
+import static java.lang.String.format;
+
+
+/*
+* An aws specific provisioning class which is using the AWS sdk to create aws instances,  and aws elastic load balance
+* */
+public class AwsProvisioner {
+
+    private final static org.apache.log4j.Logger log = org.apache.log4j.Logger.getLogger(Provisioner.class);
+
+
+    //AWS specific magic strings
+    public static final String AWS_RUNNING_STATE  = "running";
+    public static final String AWS_SYSTEM_STATUS_OK = "ok";
+    public static final String AWS_PUBLIC_IP_FILTER = "ip-address";
+    public static final String AWS_KET_NAME_FILTER = "key-name";
+
+
+    //the file wich will hole the public domain name of the created load balance
+    public static final String AWS_ELB_FILE_NAME = "aws-elb.txt";
+
+    private static final int SLEEPING_MS = 1000 * 30;
+    private static final int MAX_SLEEPING_ITTERATIONS = 12;
+
+
+    private AWSCredentials credentials;
+    private AmazonEC2 ec2;
+    private AmazonElasticLoadBalancingClient elb;
+
+    private StabilizerProperties props = new StabilizerProperties();
+
+
+    private final File agentsFile = new File("agents.txt");
+    private final File elbFile = new File(AWS_ELB_FILE_NAME);
+
+    private File credentialsFile;
+
+    private String securityGroup;
+
+    private String awsKeyName;
+    private String awsAmi;
+    private String awsBoxId;
+    private String subNetId;
+
+    private String elbProtocol;
+    private int elbPortIn;
+    private int elbPortOut;
+    private String elbAzs;
```

Comment:
can you rename this field to elbAvailabilityZones? Otherwise people need to keep a dictionary around.